app/js/lib/foxbox/foxbox.js

```diff
@@ -123,49 +127,66 @@ export default class Foxbox extends Service {
    * @returns {Promise}
    * @private
    */
-  _initDiscovery() {
+  _initDiscovery(timeout) {
     // For development purposes if you want to skip the
     // discovery phase set the 'foxbox-skipDiscovery' variable to
     // 'true'.
     if (this[p.settings].skipDiscovery) {
       return Promise.resolve();
     }
 
-    return new Promise((resolve) => {
-      this[p.net].fetchJSON(this[p.settings].registrationService)
-        .then(boxes => {
-          if (!Array.isArray(boxes) || boxes.length === 0) {
-            return resolve();
-          }
-
-          // We filter out boxes registered more than 5 minutes ago.
-          const now = Math.floor(Date.now() / 1000) - 60 * 5;
-          this[p.boxes] = Object.freeze(
-            boxes
-              .filter(box => box.timestamp - now >= 0)
-              .map(box => {
-                  // NOTE(sgiles): There is consideration to allow
-                  // only "local_origin" and "tunnel_origin", removing the need
-                  // to parse message - this merges the relevant message fields
-                  // into the main object
-                  const { local_origin, tunnel_origin } =
-                    JSON.parse(box.message);
-                  box.local_origin  = local_origin;
-                  box.tunnel_origin = tunnel_origin;
-                  return Object.freeze(box);
-              })
-          );
-
-          if (!this[p.settings].configured) {
-            this.selectBox();
-          }
-          resolve();
-        })
-        .catch(() => {
-          // When something goes wrong, we still want to resolve the promise so
-          // that a hostname set previously is reused.
-          resolve();
-        });
+    return new Promise(resolve => {
+      setTimeout(() => {
+        this[p.net].fetchJSON(this[p.settings].registrationService)
+          .then(boxes => {
+            if (!Array.isArray(boxes)) {
+              return resolve();
```

Comment:
In my understanding, in this case, we get a response from the registration server but the response is bad (likely a bug in the registration server), so there's probably no need to try again.
@ferjm: What do you think?